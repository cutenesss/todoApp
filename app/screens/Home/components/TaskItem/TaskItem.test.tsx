import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react-native';
import { TaskItem } from '.';

describe("<TaskItem />", () => {
    const mockOnPress = jest.fn();
    const defaultProps = {
        item: {
            id: 123,
            name: 'name',
            description: 'description'
        },
        onPress: mockOnPress
    }
    test('renders correctly', () => {
        const tree = renderer.create(<TaskItem {...defaultProps} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('onPress works correctly', () => {
        const itemTask = render(<TaskItem {...defaultProps} />);
        fireEvent.press(itemTask.getByTestId("TaskItem_123"));
        expect(mockOnPress).toBeCalledTimes(1);
    });
});