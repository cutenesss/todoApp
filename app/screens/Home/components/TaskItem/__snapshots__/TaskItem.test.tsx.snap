// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskItem /> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#fff",
      "borderRadius": 10,
      "elevation": 5,
      "margin": 10,
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 3.84,
    }
  }
  testID="TaskItem_123"
>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000",
        "fontSize": 15,
      }
    }
  >
    Name: name - Not complete
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000",
        "fontSize": 13,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    Description: 
    description
  </Text>
  <Text
    numberOfLines={1}
    style={
      Object {
        "color": "#000",
        "fontSize": 13,
        "fontWeight": "bold",
        "marginTop": 10,
      }
    }
  >
    Priority level: 
    Normal
  </Text>
</View>
`;
